pkg/machine/qemu/machine.go

```diff
@@ -647,6 +648,35 @@ func getDevNullFiles() (*os.File, *os.File, error) {
 	return dnr, dnw, nil
 }
 
+// qemuProcess returns the os.Process of the running QEMU instance.
+// May be nil.
+func (v *MachineVM) qemuProcessRunning() (bool, int, error) {
+	pidData, err := os.ReadFile(v.VMPidFilePath.GetPath())
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return false, -1, nil
+		}
+		return false, -1, err
+	}
+	if len(pidData) == 0 {
+		return false, -1, nil
+	}
+
+	pid, err := strconv.Atoi(strings.TrimRight(string(pidData), "\n"))
+	if err != nil {
+		logrus.Warnf("Reading QEMU pidfile: %v", err)
```

Comment:
Why is this a warning? It makes no sense to try to use pid after that if the Atoi call failed as the pid will be 0 in such case.

Reply:
Nice catch.  Needs a return.